src/game/overlay.c

```diff
@@ -214,98 +278,114 @@ void Overlay_Init()
     for (int i = 0; i < MAX_PICKUPS; i++) {
         m_Pickups[i].duration = 0;
     }
+
+    Overlay_SetupHealthBar();
+    Overlay_SetupAirBar();
+    Overlay_SetupEnemyBar();
+}
+
+void Overlay_SetHealtBarTimer(int16_t health_bar_timer)
+{
+    m_HealthBar.timer = health_bar_timer;
+}
+
+void Overlay_TimerTick()
+{
+    m_HealthBar.timer--;
+    if (m_HealthBar.timer < 0) {
+        Overlay_SetHealtBarTimer(0);
+    }
 }
 
 void Overlay_DrawHealthBar()
 {
     static int32_t old_hit_points = 0;
 
-    for (int i = 0; i < 6; i++) {
+    for (int32_t i = 0; i < 6; i++) {
         m_BarOffsetY[i] = 0;
     }
 
-    int hit_points = g_LaraItem->hit_points;
-    if (hit_points < 0) {
-        hit_points = 0;
-    } else if (hit_points > LARA_HITPOINTS) {
-        hit_points = LARA_HITPOINTS;
+    m_HealthBar.value = g_LaraItem->hit_points;
+    if (m_HealthBar.value < 0) {
+        m_HealthBar.value = 0;
+    } else if (m_HealthBar.value > m_HealthBar.max_value) {
+        m_HealthBar.value = m_HealthBar.max_value;
     }
 
-    if (old_hit_points != hit_points) {
-        old_hit_points = hit_points;
-        g_HealthBarTimer = 40;
+    if (old_hit_points != m_HealthBar.value) {
+        old_hit_points = m_HealthBar.value;
+        Overlay_SetHealtBarTimer(40);
```

Comment:
I think it's OK to just use `m_HealthBar.timer` directly here, since it's just a simple assignment, but no strong feelings about it.

Reply:
I wasn't sure for this either. I decided to use the function because every other time it's set uses that. But local in the function it could be the simple assignment since that's how the rest of it is set I guess.